src/components/SearchBox.vue

```diff
@@ -148,6 +149,14 @@ export default {
           this.$emit('update:modelValue', this.model);
           this.state.refine(this.model);
         }
+
+        // we return the local value if the input is focused to avoid
+        // concurrent updates when typing
+        const { searchInput } = this.$refs;
+        if (searchInput && searchInput.isFocused()) {
+          return this.localValue;
```

Comment:
the localValue already should be equal to the model, no?

Reply:
From what I understand, only in controlled mode. `localValue` is directly tied to the setter of `currentRefinement`, so it made sense to return its value early in the getter when the input is focused.